File: ChatGPT/Patrón de Diseño Singleton/python/singleton.py
class Singleton:
    _instance = None
    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Singleton, cls).__new__(cls)
        return cls._instance

if __name__ == "__main__":
    s1 = Singleton()
    s2 = Singleton()
    print("ID instancia s1:", id(s1))
    print("ID instancia s2:", id(s2))
    print("¿Misma instancia?", s1 is s2)

File: ChatGPT/Patrón de Diseño Singleton/ruby/singleton.rb
class Singleton
  @@instance = nil
  def self.instance
    @@instance ||= new
  end
  private_class_method :new
end

s1 = Singleton.instance
s2 = Singleton.instance
puts "ID instancia s1: #{s1.object_id}"
puts "ID instancia s2: #{s2.object_id}"
puts "¿Misma instancia? #{s1.equal?(s2)}"
